contracts/Marketplace.sol

```diff
@@ -1,74 +1,272 @@
-pragma solidity ^0.4.24;
-
-import "openzeppelin-solidity/contracts/utils/Address.sol";
-import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
-import "openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
-
-contract Service is Ownable, Pausable {
-    bytes32 public sid;
-    bytes32 public latest;
-    mapping (bytes32 => Version) public versions;
-    mapping (bytes32 => mapping(address => bool)) private _versionAccess;
-
-    struct Version {
-        bytes32 id;
-        bytes32 location;
-        uint256 price;
-        bool _exists;
-    }
+pragma solidity >=0.5.0 <0.6.0;
 
-    event VersionCreated(bytes32 sid, bytes32 id, bytes32 location, uint256 price);
+import "../node_modules/openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "../node_modules/openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
 
-    constructor(address creator, bytes32 _sid) Ownable() Pausable() public {
-        sid = _sid;
-        transferOwnership(creator);
-        addPauser(creator);
-        renouncePauser();
-    }
+contract Marketplace is Ownable, Pausable {
+  // ------------------------------------------------------
+  // Structures
+  // ------------------------------------------------------
 
-    function createVersion(bytes32 versionHash, bytes32 location, uint256 price) public onlyOwner whenNotPaused {
-        require(!versions[versionHash]._exists, "This version already exists");
-        versions[versionHash] = Version({
-            id: versionHash,
-            location: location,
-            price: price,
-            _exists: true
-        });
-        latest = versionHash;
-        emit VersionCreated(sid, versionHash, location, price);
-        return;
-    }
+  struct Service {
+    address owner;
+    string sid;
```

Comment:
sid should be `bytes20` because we use sha1 which has 160 bits.

Reply:
👍

Reply:
sid is not hashed. only hash.

Reply:
i'm replacing it by bytes

Reply:
ok, do we want to limit sid somehow? I think for sid bytes32 will be just fine, but we have to limit it on core side - we already have limitation that len(sid) != len(hash)

Reply:
how many ascii encoded characters in a bytes20?

Reply:
asci is saved on 1 byte == 8 bit
so 20 ascii chars